app/services/PaymentHistoryService.scala

```diff
@@ -35,43 +35,28 @@ import scala.concurrent.{ExecutionContext, Future}
 class PaymentHistoryService @Inject()(repaymentHistoryConnector: RepaymentHistoryConnector,
                                       financialDetailsConnector: FinancialDetailsConnector,
                                       implicit val dateService: DateServiceInterface,
-                                      val appConfig: FrontendAppConfig)
+                                      implicit val appConfig: FrontendAppConfig)
                                      (implicit ec: ExecutionContext) {
 
-  @deprecated("Use getPaymentHistoryV2 instead", "MISUV-8845")
-  def getPaymentHistory(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[PaymentHistoryError.type, List[Payment]]] = {
+  def getPaymentHistory(implicit hc: HeaderCarrier, user: MtdItUser[_]): Future[Either[PaymentHistoryError.type, Seq[Payment]]] = {
 
-    val orderedTaxYears: List[Int] = user.incomeSources.orderedTaxYearsByYearOfMigration.reverse.take(appConfig.paymentHistoryLimit)
+    val listOfCalls = user.incomeSources.orderedTaxYearsInWindows
 
-    Future.sequence(orderedTaxYears.map(year => financialDetailsConnector.getPayments(TaxYear(year-1, year)))) map { paymentResponses =>
-      val paymentsContainsFailure: Boolean = paymentResponses.exists {
-        case Payments(_) => false
-        case PaymentsError(status, _) if status == NOT_FOUND => false
-        case PaymentsError(_, _) => true
+    Future.sequence(listOfCalls.map { years =>
+      val (from, to) = (years.min, years.max)
+      Logger("application").debug(s"Getting payment history for TaxYears: ${from} - ${to}")
+
+      for {
+        response <- financialDetailsConnector.getPayments(TaxYear.forYearEnd(from), TaxYear.forYearEnd(to))
+      } yield response match {
+        case Payments(payments) => Right(payments.distinct)
```

Comment:
not sure if we should do any data processing for payments, how "distinct" function would work??